api/envoy/extensions/http/ext_proc/response_processors/save_processing_response/v3/save_processing_response.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package envoy.extensions.http.ext_proc.response_processors.save_processing_response.v3;
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.http.ext_proc.response_processors.save_processing_response.v3";
+option java_outer_classname = "SaveProcessingResponseProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/http/ext_proc/response_processors/save_processing_response/v3;save_processing_responsev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Save Processing Response from external processor.]
+// [#extension: envoy.http.ext_proc.response_processors.save_processing_response]
+
+// Extension to save the :ref:`response
+// <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse>` from the external processor as
+// filter state with name
+// "envoy.http.ext_proc.response_processors.save_processing_response[.:ref:`filter_state_name_suffix
+// <envoy_v3_api_field_extensions.http.ext_proc.response_processors.save_processing_response.v3.SaveProcessingResponse.filter_state_name>`].
+// [#next-free-field: 11]
+message SaveProcessingResponse {
+  string filter_state_name_suffix = 1;
+
+  // Save the response to filter state when :ref:`request_headers
+  // <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.request_headers>` is set.
+  bool save_request_headers = 2;
+
+  // Save the response to filter state when :ref:`response_headers
+  // <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.response_headers>` is set.
+  bool save_response_headers = 3;
+
+  // Save the response to filter state when :ref:`request_body
+  // <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.request_body>` is set.
+  bool save_request_body = 4;
+
+  // Save the response to filter state when :ref:`response_body
+  // <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.response_body>` is set.
+  bool save_response_body = 5;
+
+  // Save the response to filter state when :ref:`request_trailers
+  // <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.request_trailers>` is set.
+  bool save_request_trailers = 6;
+
+  // Save the response to filter state when :ref:`response_trailers
+  // <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.response_trailers>` is set.
+  bool save_response_trailers = 8;
+
+  // Save the response to filter state when :ref:`immediate_response
+  // <envoy_v3_api_field_service.ext_proc.v3.ProcessingResponse.immediate_response>` is set.
+  bool save_immediate_response = 9;
+
+  // When false, skips saving the response if there was an error when processing
```

Comment:
nit: might be better to describe on the positive side (If true...) as the default is false.

Reply:
Done.